lib/graph3d/Range.js

```diff
@@ -0,0 +1,84 @@
+////////////////////////////////////////////////////////////////////////////////
+// class Range
+//
+// Helper class to make working with related min and max values easier.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+
+function Range() {
+  this.min = undefined;
+  this.max = undefined;
+}
+
+
+Range.prototype.add = function(value) {
+  if (value === undefined) return;
+
+  if (this.min === undefined || this.min > value ) {
+    this.min = value;
+  }
+
+  if (this.max === undefined || this.max < value) {
+    this.max = value;
+  }
+};
+
+
+/**
+ * Adjust the current range so that the passed range fits in it.
+ */
+Range.prototype.combine = function(range) {
+   this.add(range.min);
+   this.add(range.max);
+};
+
+
+/**
+ * Replace the current min and max values with the
+ * passed values.
+ */
+Range.prototype.override = function(min, max) {
+  if (min !== undefined) {
+    this.min = min;
+  }
+
+  if (max !== undefined) {
+    this.max = max;
+  }
+
+  if (this.max <= this.min) this.max = this.min + 1;
```

Comment:
The reason of the existence of this line, in fact this whole method, is compatibility with previous working. I'm careful not to disrupt the previous implementation.

But to be honest, it's a bit kludgy and has nothing to do with actual min/max ranges. I'll consider doing this differently.